src/main/java/org/broadinstitute/consent/http/resources/DatasetResource.java

```diff
@@ -203,6 +206,89 @@ public Response createDatasetRegistration(
     }
   }
 
+  @PUT
+  @Consumes({MediaType.MULTIPART_FORM_DATA})
+  @Produces({MediaType.APPLICATION_JSON})
+  @Path("/study/{studyId}")
+  @RolesAllowed({ADMIN, CHAIRPERSON, DATASUBMITTER})
+  /*
+   * This endpoint accepts a json instance of a dataset-registration-schema_v1.json schema.
+   * With that object, we can fully update the study/datasets from the provided values.
+   */
+  public Response updateStudyByRegistration(
+      @Auth AuthUser authUser,
+      FormDataMultiPart multipart,
+      @PathParam("studyId") Integer studyId,
+      @FormDataParam("dataset") String json) {
+    try {
+      User user = userService.findUserByEmail(authUser.getEmail());
+      Study existingStudy = datasetRegistrationService.findStudyById(studyId);
+
+      // Manually validate the schema from an editing context. Validation with the schema tools
+      // enforces it in a creation context but doesn't work for editing purposes.
+      Gson gson = GsonUtil.gsonBuilderWithAdapters().create();
+      DatasetRegistrationSchemaV1 registration = gson.fromJson(json, DatasetRegistrationSchemaV1.class);
+
+      // Not modifiable: Study Name, Data Submitter Name/Email, Primary Data Use,
+      // Secondary Data Use
+      if (Objects.nonNull(registration.getStudyName()) && !registration.getStudyName().equals(existingStudy.getName())) {
+        throw new BadRequestException("Invalid change to Study Name");
+      }
+      if (Objects.nonNull(registration.getDataSubmitterUserId()) &&!registration.getDataSubmitterUserId().equals(existingStudy.getCreateUserId())) {
+        throw new BadRequestException("Invalid change to Data Submitter");
+      }
+      // Data use changes are not allowed for existing datasets
+      List<ConsentGroup> invalidConsentGroups = registration.getConsentGroups()
+          .stream()
+          .filter(cg -> Objects.nonNull(cg.getDatasetId()))
+          .filter(cg ->
+              Objects.nonNull(cg.getOpenAccess()) ||
+                  Objects.nonNull(cg.getGeneralResearchUse()) ||
+                  Objects.nonNull(cg.getHmb()) ||
+                  (Objects.nonNull(cg.getDiseaseSpecificUse()) && cg.getDiseaseSpecificUse().size() > 0) ||
+                  Objects.nonNull(cg.getPoa()) ||
+                  Objects.nonNull(cg.getOtherPrimary()) ||
+                  Objects.nonNull(cg.getNmds()) ||
+                  Objects.nonNull(cg.getGso()) ||
+                  Objects.nonNull(cg.getPub()) ||
+                  Objects.nonNull(cg.getCol()) ||
+                  Objects.nonNull(cg.getIrb()) ||
+                  Objects.nonNull(cg.getGs()) ||
+                  Objects.nonNull(cg.getMor()) ||
+                  Objects.nonNull(cg.getMorDate()) ||
+                  Objects.nonNull(cg.getNpu()) ||
+                  Objects.nonNull(cg.getOtherSecondary()))
```

Comment:
nit: could we have a method on the consent group to determine whether any of these are filled?